src/styled_text.rs

```diff
@@ -57,11 +76,16 @@ impl StyledText {
     }
 }
 
-fn render_as_string(renderable: &(Style, String), prompt_style: &Style) -> String {
+fn render_as_string(
+    renderable: &(Style, String),
+    prompt_style: &Style,
+    multiline_prompt: &str,
+) -> String {
     let mut rendered = String::new();
+    let formatted_multiline_prompt = format!("\n{}", multiline_prompt);
     for (line_number, line) in renderable.1.split('\n').enumerate() {
         if line_number != 0 {
-            rendered.push_str(&prompt_style.paint("\n: ").to_string());
+            rendered.push_str(&prompt_style.paint(&formatted_multiline_prompt).to_string());
```

Comment:
Allocation `.to_string()` is unnecessary as you immediately consume them. Can be replaced by either `.deref()`, `.as_ref()`, or prepending with `&*` (which I personally find less readable).

Reply:
My bad, the deref is implemented differently and this breaks formatting